### DISCUSSION TITLE
How to implement useExpand and useGlobalFilter properly ?

### DISCUSSION THREAD

**Discussion by ZiedHf** [TanStack/table#3695] [Q&A] [answered]
_Example in CodeSandBox below_

I am trying to use `useExpand` and `useGlobalFilter` in the same table but I am facing some issues : 

1. In the example below, I don't understand why writing "injury" in the Filter Input will display it as it is a simple row ? If I provide a custom globalFilter method I can see that the row is unaxpandable `{ canExpand: false }` 

![image](https://user-images.githubusercontent.com/7974028/153068742-4052c753-ecef-46dc-849b-214801642862.png)

![image](https://user-images.githubusercontent.com/7974028/153068937-8dc4bfa4-9a43-474e-b735-a59cbe9e4999.png)

Even after deleting the global filter ?!
![image](https://user-images.githubusercontent.com/7974028/153068970-e17ba918-1575-4a4f-8ce6-72c260f93cde.png)

What I am doing wrong ?
How can I display it as a Parent Row after feletring and how to display all of its Sub Rows ?

This is what I am looking for :

- ![image](https://user-images.githubusercontent.com/7974028/153069293-cc600c76-8b09-4823-ae40-6624860e5bcc.png)

- ![image](https://user-images.githubusercontent.com/7974028/153069435-88e1b25b-283d-4c01-b293-5bd9d400f71a.png)

- ![image](https://user-images.githubusercontent.com/7974028/153069510-cf487fe2-d438-4973-a9c9-6cfe2a4d7e23.png)

2. In a custom globalFilter method, is it possible to display a row according to the state of its Parent or the state of its SubRows ?
3. What is the best way to change the IsExpanded state of a row after filtering ? Should this go into the globalFilter method as well ?

A CodeSandBox example :
https://codesandbox.io/s/react-table-expanded-global-filter-l2bct

**Comment by ZiedHf** [comment]
A fix for this 

> Even after deleting the global filter ?!
> ![image](https://user-images.githubusercontent.com/7974028/153068970-e17ba918-1575-4a4f-8ce6-72c260f93cde.png)

[Here](https://github.com/TanStack/react-table/issues/2064#issuecomment-623199018)

**Comment by ZiedHf** [comment, marked as answer]
Fix 

https://codesandbox.io/s/react-table-expanded-global-filter-1-kibuy

**Reply by vpak555** [reply]
That is awesome. Helped a lot! You are the only one solved and posted this solution. Thanks a lot!
Also interested how to manipulate expansion/collapse of rows and subrows depending on filter value?
For example, if filter value is matching 3rd level of data then expand all 3 levels, if it's matching second level, then expand 2 levels, but hide 3rd level etc.
Would appreciate any suggestion!

**Comment by NarenderDhamarlapally** [comment]
please give this in v8

**Comment by mbauchet** [comment]
Is there any working codesandbox for this?
